homeassistant/components/zwave_js/sensor.py

```diff
@@ -208,10 +199,15 @@ def async_add_sensor(info: ZwaveDiscoveryInfo) -> None:
         """Add Z-Wave Sensor."""
         entities: list[ZWaveBaseEntity] = []
 
+        if info.platform_data:
+            entity_description_key, unit = info.platform_data
+        else:
+            entity_description_key, unit = None, None
         entity_description = ENTITY_DESCRIPTION_KEY_MAP.get(
-            info.platform_data
+            entity_description_key
         ) or ZwaveSensorEntityDescription("base_sensor")
         entity_description.info = info
+        entity_description.native_unit_of_measurement = unit
```

Comment:
I guess it's the same problem as above. It's not really good to do it like this.

Reply:
do you have an alternate suggestion? I don't think we want to create SensorEntityDescription's for every combo of options do we?

Reply:
ignore what I said above, I see your suggestion in the previous comment

Reply:
lmk what you think about the current implementation